app/forms/provider_onboarding_form.rb

```diff
@@ -12,26 +12,26 @@ def onboard
   end
 
   def provider
-    @provider ||= Provider.find_by(code:)
+    @provider ||= Provider.find_by_id(id)
```

Comment:
I was under the impression `find_by_attributes` methods we deprecated back in Rails 5. Shouldn't this be `find(id)`, as you don't need to specify `id` and `id` should be present.

Reply:
I kept the find_by pattern because we expect a nil or a record here https://github.com/DFE-Digital/itt-mentor-services/blob/92461ea900f3cc52f89749ecbcf03e0c1530d3e9/app/forms/provider_onboarding_form.rb#L21

`find` would return us an error if we can't find it. I can just rescue a `ActiveRecord::RecordNotFound` and put the message in there if you want

Reply:
Fixed 9bc10093ea8ca35cf0f036cd52a556352a202e45